### PR TITLE
Use new style caffe2 targets

### PR DESCRIPTION
**Pull request by Yangqing**
This basically allows Detectron to be built without having to include the many cmake files from Caffe2. Caffe2 now properly exports utility functions and dependent libraries, and so one can use the new style targets "caffe2_library" and "caffe2_gpu_library" directly.

This should also solve problems like #251 after we update the conda version. cc @pjh5.

### CODE REVIEW

**Comment by Yangqing** [comment]
Also cc @orionr.

File: lib/cmake/legacy/legacymake.cmake [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+     ${CUSTOM_OPS_CPU_SRCS})
+
+target_include_directories(
+    caffe2_detectron_custom_ops PRIVATE
+    ${CAFFE2_INCLUDE_DIRS})
+target_link_libraries(caffe2_detectron_custom_ops caffe2)
+install(TARGETS caffe2_detectron_custom_ops DESTINATION lib)
+
+# Install custom GPU ops lib.
+if (HAVE_CUDA)
```
Comment (pjh5) [inline lib/cmake/legacy/legacymake.cmake]: Should this be ${HAVE_CUDA} ?

**Comment by rbgirshick** [comment]
@ir413 if this look good to you, please merge.

File: lib/cmake/legacy/legacymake.cmake (hunk 1 above)
Comment (Yangqing) [reply lib/cmake/legacy/legacymake.cmake]: Went a bit sloppy, I think both would work...

**Pull request closed without merging**